manifests/osfamily/suse.pp

```diff
@@ -5,104 +5,119 @@
+        }
 
-      file { $legacy_gpg_path:
-        ensure => present,
-        owner  => 0,
-        group  => 0,
-        mode   => '0644',
-        source => "puppet:///modules/puppet_agent/${legacy_keyname}",
-      }
+        file { $legacy_gpg_path:
+          ensure => present,
+          owner  => 0,
+          group  => 0,
+          mode   => '0644',
+          source => "puppet:///modules/puppet_agent/${legacy_keyname}",
+        }
 
-      # Given the path to a key, see if it is imported, if not, import it
-      $legacy_gpg_pubkey = "gpg-pubkey-$(echo $(gpg --homedir ${gpg_homedir} --with-colons ${legacy_gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
-      $gpg_pubkey        = "gpg-pubkey-$(echo $(gpg --homedir ${gpg_homedir} --with-colons ${gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
+        # Given the path to a key, see if it is imported, if not, import it
+        $legacy_gpg_pubkey = "gpg-pubkey-$(echo $(gpg --homedir ${gpg_homedir} --with-colons ${legacy_gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
+        $gpg_pubkey        = "gpg-pubkey-$(echo $(gpg --homedir ${gpg_homedir} --with-colons ${gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr '[:upper:]' '[:lower:]'))"
 
-      exec { "import-${legacy_keyname}":
-        path      => '/bin:/usr/bin:/sbin:/usr/sbin',
-        command   => "rpm --import ${legacy_gpg_path}",
-        unless    => "rpm -q ${legacy_gpg_pubkey}",
-        require   => File[$legacy_gpg_path],
-        logoutput => 'on_failure',
-      }
+        exec { "import-${legacy_keyname}":
+          path      => '/bin:/usr/bin:/sbin:/usr/sbin',
+          command   => "rpm --import ${legacy_gpg_path}",
+          unless    => "rpm -q ${legacy_gpg_pubkey}",
+          require   => File[$legacy_gpg_path],
+          logoutput => 'on_failure',
+        }
 
-      exec { "import-${keyname}":
-        path      => '/bin:/usr/bin:/sbin:/usr/sbin',
-        command   => "rpm --import ${gpg_path}",
-        unless    => "rpm -q ${gpg_pubkey}",
-        require   => File[$gpg_path],
-        logoutput => 'on_failure',
-      }
+        exec { "import-${keyname}":
+          path      => '/bin:/usr/bin:/sbin:/usr/sbin',
+          command   => "rpm --import ${gpg_path}",
+          unless    => "rpm -q ${gpg_pubkey}",
+          require   => File[$gpg_path],
+          logoutput => 'on_failure',
+        }
 
-      if getvar('::puppet_agent::manage_repo') == true {
-        # Set up a zypper repository by creating a .repo file which mimics a ini file
-        $repo_file = '/etc/zypp/repos.d/pc_repo.repo'
-        $repo_name = 'pc_repo'
-        $agent_version = $::puppet_agent::package_version
+        if getvar('::puppet_agent::manage_repo') == true {
```

Comment:
SUSE is similar to RHEL. Sets up the keys but stops at the last mile.

Again, not sure if the keys are necessary if we aren't managing the repo—unless I am completely missing something.

All of this `manage_repo` business is admittedly somewhat out-of-scope for MODULES-6604. But it is something that has bothered me for a long time about this module.  (I'd be glad to submit another PR in the future if makes more sense to tackle it later.)